### PR TITLE
Tools and front

### PR DESCRIPTION
**Pull request by zxan1285**

### CODE REVIEW

File: server/lib/logTypes.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
+module.exports = {
```
Comment (sandrinodm) [inline server/lib/logTypes.js]: Remove this. We should be able to get this from log extension tools

File: server/lib/processLogs.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      };
+
+      Object.keys(log).forEach((key) => {
+        data[key] = log[key];
+      });
+
+      data[index] = log[index] || 'auth0';
+      data.message = JSON.stringify(log);
+
+      const url = config('LOGSTASH_TOKEN') ? `${config('LOGSTASH_URL')}?token=${config('LOGSTASH_TOKEN')}` : config('LOGSTASH_URL');
```
Comment (sandrinodm) [inline server/lib/processLogs.js]: Logic needs to be improved. If url ealready contains `?`, you should add `&`

File: webtask.json [hunk 3]
```diff
@@ -27,6 +30,38 @@
...
+      "type": "password"
+    },
+    "START_FROM": {
+      "description": "CheckpointId of log to start with."
+    },
+    "SLACK_INCOMING_WEBHOOK_URL": {
+      "description": "Slack webhook"
+    },
+    "SLACK_SEND_SUCCESS": {
+      "description": "Send success messages to slack?",
```
Comment (sandrinodm) [inline webtask.json]: description: "If disabled, only failed runs will be reported to Slack"

File: webtask.json [hunk 4]
```diff
@@ -27,6 +30,38 @@
...
+      "type": "select",
+      "allowMultiple": true,
```
Comment (sandrinodm) [inline webtask.json]: false

File: webtask.json [hunk 5]
```diff
@@ -27,6 +30,38 @@
...
+      "options": [
+        {
+          "value": "No",
```
Comment (sandrinodm) [inline webtask.json]: value: "false", text: "No"

File: webtask.json [hunk 6]
```diff
@@ -27,6 +30,38 @@
...
+          "text": ""
+        },
+        {
+          "value": "Yes",
```
Comment (sandrinodm) [inline webtask.json]: value: "true", text: "Yes"

File: server/lib/processLogs.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      if (!logs || !logs.length) {
+        return callback();
+      }
+
+      logger.info(`Sending ${logs.length} logs to Logstash.`);
+
+      async.eachLimit(logs, 100, sendLog, callback);
+    };
+
+    const slack = new loggingTools.reporters.SlackReporter({ hook: config('SLACK_INCOMING_WEBHOOK_URL'), username: 'auth0-logs-to-logstash', title: 'Logs To Logstash' });
```
Comment (sandrinodm) [inline server/lib/processLogs.js:79]: What about SLACK_SEND_SUCCESS ?

Comment (zxan1285) [reply server/lib/processLogs.js:79]: Checking on line [96](https://github.com/auth0/auth0-logs-to-logstash/pull/4/files#diff-c7cebf846964f0fdc867268be3906c1cR96)

File: server/lib/processLogs.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+const loggingTools = require('auth0-log-extension-tools');
+
+const config = require('../lib/config');
+const logger = require('../lib/logger');
+
+module.exports = (storage) =>
+  (req, res, next) => {
+    const wtBody = (req.webtaskContext && req.webtaskContext.body) || req.body || {};
+    const wtHead = (req.webtaskContext && req.webtaskContext.headers) || {};
+    const isCron = (wtBody.schedule && wtBody.state === 'active') || (wtHead.referer === 'https://manage.auth0.com/' && wtHead['if-none-match']);
```
Comment (sandrinodm) [inline server/lib/processLogs.js]: use this as referrer:`config('AUTH0_MANAGE_URL') + '/'`

**Review by sandrinodimattia — APPROVED** [review]

**Pull request merged**